crates/rrg/src/action/execute_signed_command.rs

```diff
@@ -0,0 +1,568 @@
+    /// Standard output of the command executiom.
+    stdout: Vec<u8>,
+    /// Wheather standard output is truncated.
+    truncated_stdout: bool,
+    /// Standard error of the command executiom.
+    stderr: Vec<u8>,
+    /// Wheather stderr is truncated.
+    truncated_stderr: bool,
+}
+
+enum Stdin {
+    None,
+    Unsigned(Vec<u8>),
+    Signed(Vec<u8>),
+}
+
+/// An error indicating that the command signature is missing.
+#[derive(Debug)]
+struct MissingCommandSignatureError;
+
+impl std::fmt::Display for MissingCommandSignatureError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write! {
+            fmt,
+            "missing command signature"
+        }
+    }
+}
+
+impl std::error::Error for MissingCommandSignatureError {}
+
+/// An error indicating that stdin of the command couln't be captured.
+#[derive(Debug)]
+struct CommandStdinError;
+
+impl std::fmt::Display for CommandStdinError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write! {
+            fmt,
+            "failed to capture stdin when spawning the command process"
+        }
+    }
+}
+
+impl std::error::Error for CommandStdinError {}
+
+/// Handles invocations of the `execute_signed_command` action.
+pub fn handle<S>(session: &mut S, mut args: Args) -> crate::session::Result<()>
+where
+    S: crate::session::Session,
+{
+    match session.args().command_verification_key {
+        Some(key) => key
+            .verify_strict(&args.raw_command, &args.ed25519_signature)
+            .map_err(crate::session::Error::action)?,
+        None => return Err(crate::session::Error::action(MissingCommandSignatureError)),
+    };
+
+    let command_path = &std::path::PathBuf::try_from(args.command.take_path())
+        .map_err(crate::session::Error::action)?;
+
+    // // For calling `cmd /C <command>` in windows, C:\Windows\System32
+    // // needs to be on the PATH, otherwise <command> is not found.
+    // let filtered_env: HashMap<String, String> = env::vars()
+    //     .filter(|&(ref k, _)| k.to_lowercase() == "path")
+    //     .collect();
+
+    let mut command_process = Command::new(command_path)
+        .stdin(std::process::Stdio::piped())
+        .args(args.command.take_args())
+        .env_clear()
+        // .envs(filtered_env)
+        .envs(args.command.take_env())
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::piped())
+        .spawn()
+        .map_err(crate::session::Error::action)?;
+
+    let command_start_time = std::time::SystemTime::now();
+
+    let mut command_stdin = match command_process.stdin.take() {
+        Some(command_stdin) => command_stdin,
+        None => return Err(crate::session::Error::action(CommandStdinError)),
+    };
+
+    let handle = std::thread::spawn(move || match args.stdin {
+        Stdin::Signed(signed) => command_stdin
+            .write(&signed[..])
+            .expect("Failed to write to stdin"),
```

Comment:
Nit: expectations should not be capitalized like sentences (similarly to explicit messages passed to `panic!`).

Reply:
Done.